presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -804,11 +814,153 @@ public void testSchemaEvolution()
+                    "   CASE WHEN orderkey % 49 = 0 THEN null ELSE (comment, comment) END AS varchar_array, \n" +
+                    "'2019-11-01' AS ds \n" +
+                    "FROM lineitem) \n";
+
+            Function<String, String> rewriter = query -> query.replaceAll("info.orderkey", "info[1]")
+                    .replaceAll("info.linenumber", "info[2]")
+                    .replaceAll("info.shipdate.ship_day", "info[3][1]")
+                    .replaceAll("info.shipdate.ship_year", "info[3][3]")
+                    .replaceAll("info.shipdate", "info[3]")
+                    .replaceAll("dates\\[1\\].day", "dates[1][1]")
+                    .replaceAll("map_flags", "flags")
+                    .replaceAll("test_maps", "lineitem_ex")
+                    .replaceAll("cardinality", "array_length");
+
+            // no filter
+            assertQueryUsingH2Cte("SELECT * FROM test_file_format_orc", cte);
+            assertQueryUsingH2Cte("SELECT comment FROM test_file_format_orc", cte);
+            assertQueryUsingH2Cte("SELECT COUNT(*) FROM test_file_format_orc", cte);
+
+            //filter on partition col
+            assertQueryUsingH2Cte("SELECT comment from test_file_format_orc WHERE ds='2019-11-01'", cte);
+            assertQueryReturnsEmptyResult("SELECT comment FROM test_file_format_orc WHERE ds='2019-11-02'");
+
+            // range filters and filter functions
+            assertQueryUsingH2Cte("SELECT orderkey from test_file_format_orc WHERE orderkey < 1000", cte);
+            assertQueryUsingH2Cte("SELECT orderkey, comment from test_file_format_orc WHERE orderkey < 1000 AND comment LIKE '%final%'", cte);
+            assertQueryUsingH2Cte("SELECT COUNT(*) from test_file_format_orc WHERE orderkey < 1000", cte);
+
+            assertQueryUsingH2Cte("SELECT COUNT(*) FROM test_file_format_orc WHERE concat(ds,'*') = '2019-11-01*'", cte);
+            assertQueryUsingH2Cte("SELECT orderkey FROM test_file_format_orc WHERE comment LIKE '%final%'", cte);
+
+            assertQueryUsingH2Cte("SELECT discount FROM test_file_format_orc WHERE discount > 0.01", cte);
+            assertQueryUsingH2Cte("SELECT * FROM test_file_format_orc WHERE discount > 0.01 and discount + tax > 0.03", cte);
+            assertQueryUsingH2Cte("SELECT COUNT(*) FROM test_file_format_orc WHERE discount = 0.0", cte);
+
+            assertQueryUsingH2Cte("SELECT COUNT(*) FROM test_file_format_orc WHERE discount_real > 0.01", cte);
+            assertQueryUsingH2Cte("SELECT * FROM test_file_format_orc WHERE tax_real > 0.01 and discount_real > 0.01", cte);
+
+            assertQueryUsingH2Cte("SELECT keys FROM test_file_format_orc WHERE keys IS NOT NULL", cte);
+            assertQueryUsingH2Cte("SELECT keys FROM test_file_format_orc WHERE keys IS NULL", cte);
+            assertQueryUsingH2Cte("SELECT linenumber FROM test_file_format_orc WHERE keys[1] % 5 = 0 AND keys[2] > 100", cte, rewriter);
```

Comment:
this query shouldn't require custom rewriter; fix `assertQueryUsingH2Cte` helper method instead

```
    private void assertQueryUsingH2Cte(String query, String cte)
    {
        assertQueryUsingH2Cte(query, cte, Function.identity());
    }
```